### ISSUE TITLE
keyify: panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by Bo0mer** [dominikh/go-tools#244]
First of all thanks for you awesome work!

I'm using `keyify` from vim, but the problem reproduces from the command line. I'm seeing the following panic.

```text
$ keyify './notifications.go:#151'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x11f0e40]

goroutine 1 [running]:
go/token.(*File).Size(...)
	/usr/local/go/src/go/token/position.go:115
main.fileOffsetToPos(0x0, 0x97, 0x97, 0x52, 0xc4200a4b00, 0xc42002d180, 0x0)
	/Users/ivan/workspace/Go/src/github.com/dominikh/go-tools/cmd/keyify/position.go:56 +0x30
main.main()
	/Users/ivan/workspace/Go/src/github.com/dominikh/go-tools/cmd/keyify/keyify.go:114 +0x5fb
```
Unfortunately, the code is closed source and I'm not able to provide it. I'll try to setup a minimal reproducible example and post it here.

Cheers,
Ivan

**Comment by Bo0mer** [comment]
I've done some debugging, and it seems that the problem is around the following lines:

https://github.com/dominikh/go-tools/blob/376b3b58b9e4def403181ee2fd3d4cc7de8375ae/cmd/keyify/keyify.go#L105-L114

On line 105, the pkg gets populated to the test package (`mypackage_test`) instead of the implementation (`mypackage`). When later trying to find the file provided as an command-line argument in the set of package files, it is not there, thus is `nil`.

The documentation for `InitialPackages` states:
```text
// InitialPackages returns a new slice containing the set of initial
// packages (Created + Imported) in unspecified order.
```
Note the "unspecified order" here.

If I change the loop to go through all initial packages, the problem does not appear and the output of `keyify` is as expected.

```go
	var pkg *loader.PackageInfo
outer:
	for _, initialPackage := range lprog.InitialPackages() {
		for _, ff := range initialPackage.Files {
			file := lprog.Fset.File(ff.Pos())
			if file.Name() == name {
				pkg = initialPackage
				af = ff
				tf = file
				break outer
			}
		}
	}

```

Please let me know if any more debugging is needed.
Thanks and Happy Holidays!
Ivan

**Comment by dominikh** [comment]
Thank you for the detailed analysis. I'll likely fix the issue in the first weeks of 2018.

**Comment by dominikh** [comment]
Your analysis was exactly right. I've fixed the issue in d893811, and the fix is part of Staticcheck 2017.2.2.
